### DISCUSSION TITLE
Create/Edit Qualification plans via UI

### DISCUSSION THREAD

**Discussion by Yuri05** [Open-Systems-Pharmacology/QualificationPlan#164] [Ideas]
Users would like to have an easy possibility to create **and edit** the qualification plans with the UI support.

**Comment by PavelBal** [comment]
## Projects 
- **Add project**: Select a *.json project snapshot (locally of from GitHub-repo)
- **List of projects**: Shows the loaded projects. Each project has an Id (editable by user). Allows to remove a project.
- **Project inheritance**: For each loaded project, user can add an inheritance. "Add inheritance" button adds a new entry with:
  - Replaced entity: List of all replaceable entities (Individuals, Compounds, etc) present in the project.
  - Replace with: List of loaded projects that can be used for replacement (== They have a BB of the respective type with the same name)
- `Parameter inheritance`:
  - Either manual paths or parameter trees to select from. I do not know what exactly the feature supports, so cannot specify further.

## Observed data
- `Add observed data`: Select an excel/CSV file with observed data (locally or GitHub)

## Sections
- **Add section**: User types in Id, Title, and selects the content file (local or GitHub)
- For each section, a sub-section can be added
- **Remove section** (if sub-sections present - these are removed too)

## Inputs
- **Add input**: List of all entries from all project (useful grouping? By project/BB-type...) for which inputs can be generated
- **Remove input**

## Plots
### Settings
- All global axes and plot settings grouped by plot type, pre-filled with default values...

### All plots
  - **Add simulation**: List of all simulations in from all projects. User can define the `SectionId` (maybe from the list of section that are specified?)
 
### GOFMergedPlots
  - **Add new GOF plot**:
    - Title
    - SectionId (from the list of available sections)
    - Artifacts: check-boxes
    - PlotType: check-boxes
    - For each GOF plot:
      - Add group:
        - User defines a caption
        - Symbol selection from pre-defined set
        - Add output mapping:
          - Selection of all available model outputs, with option to manually type in the output path (optionally - group by projects -> Simulations)
          - Selection of all available observed data (optionally: show only data available on the project of model output AND the data sets defined in "Observed data" section)
          - Selection of color for each entry (color palette)


### Comparison time profiles plots
  - **Add new time profile plot**:
    - Title
    - SectionId (from the list of available sections)
    - For each plot:
      - Add mapping:
        - User defines a caption
        - Symbol selection from pre-defined set
        - Add output mapping:
          - Selection of all available model outputs, with option to manually type in the output path (optionally - group by projects -> Simulations)
          - Selection of all available observed data (optionally: show only data available on the project of model output AND the data sets defined in "Observed data" section)
          - Selection of color for each entry (color palette)
          - Not sure about "Simulation duration", maybe setting axes limits would suffice.. don't know


That's it for now. Lots of additional improvements could be done, but also such a GUI would be a MASSIVE help.


Additional specs:
  - Loading an evaluation/qualification plan
  - Checking the loaded plan (i.e. the referenced files exist, the models in the projects exist, data exist etc)
